controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -323,6 +323,88 @@ function readInputData() {
 		));
 	}
 
+	/**
+	 * @copydoc Form::validate()
+	 */
+	function validate($callHooks = true) {
+		// Display error if several blind participants are selected (by the editor) for a review discussion:
+		// 1) author and one or several blind reviewers
+		// 2) several blind reviewers.
+		// If only several authors are selected, the error will not be displayed.
+		// Editors are ignored.
+		$query = $this->getQuery();
+		// Queryies only support ASSOC_TYPE_SUBMISSION so far (see above)
+		if ($query->getAssocType() == ASSOC_TYPE_SUBMISSION) {
+			$request = Application::get()->getRequest();
+			$user = $request->getUser();
+			$submissionId = $query->getAssocId();
+
+			// Get current users roles
+			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
+			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+			$assignedRoles = [];
+			$usersAssignments = $stageAssignmentDao->getBySubmissionAndStageId($submissionId, $query->getStageId(), null, $user->getId());
+			while ($usersAssignment = $usersAssignments->next()) {
+				$userGroup = $userGroupDao->getById($usersAssignment->getUserGroupId());
+				$assignedRoles[] = $userGroup->getRoleId();
+			}
+
+			// if current user is editor, forbid several blind participants (reviewers and author) in a review discussion
+			if ($user->hasRole([ROLE_ID_SITE_ADMIN], CONTEXT_SITE) ||
+				array_intersect([ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR], $assignedRoles)) {
```

Comment:
I think this validation should occur always, not only when the user is an editor. Because a malicious user could still submit the form with any participant IDs, not just the ones that the UI shows to them.